subprojects/language-native/src/main/java/org/gradle/language/cpp/plugins/CppLibraryPlugin.java

```diff
@@ -117,6 +131,14 @@ public File call() throws Exception {
         });
         apiElements.getOutgoing().artifact(publicHeaders);
 
+        final Usage swiftApiUsage = objectFactory.named(Usage.class, Usage.SWIFT_API);
+        final Configuration swiftApiElements = configurations.maybeCreate("swiftApiElements");
+        swiftApiElements.extendsFrom(library.getApiDependencies());
+        swiftApiElements.setCanBeResolved(false);
+        swiftApiElements.getAttributes().attribute(Usage.USAGE_ATTRIBUTE, swiftApiUsage);
+        swiftApiElements.getOutgoing().artifact(library.getModuleMapFile());
+        swiftApiElements.getOutgoing().artifact(publicHeaders);
```

Comment:
Another weird thing, the modulemap files aren't a Swift specific thing.  We could theoretically build only C++ and want to use modulemaps to modularize our software. I may never use a Swift plugin in my project, but I still get Swift stuff in my project.

And since we're extending from `api`, I think that means we wouldn't see the modulemaps defined for transitive dependencies?

e.g.:

Swift app -> C++ module A (api dependency) -> C++ module B

If the Swift app needs to import something in `B` to use with `A`, I don't think they can without explicitly depending on B?

@adammurdoch is there some way to use the existing variant information here and include the optional modulemap when a C++ library is used by a Swift component?

Reply:
Per our conversations, this has been moved to the consumer side and is implemented with an artifact schema to allow c++ usage when something has been added to a configuration with swift usage.

Transitive module map dependencies are handled by this as well.